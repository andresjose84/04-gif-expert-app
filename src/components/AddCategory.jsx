import React, { useState } from 'react';
import PropTypes from 'prop-types';

const AddCategory = ({ setCategories }) => {
    const [inputState, setInputState] = useState('');

    const handleOnChange = (e) => {
        //console.log(e.target.value);
        setInputState(e.target.value);
    };

    const handleSubmit = (e) => {
        e.preventDefault();
        if (inputState.trim().length > 1) {
            console.log('Submit hecho', e.target);
            setCategories(cats => [inputState, ...cats]);
            setInputState('');
        }
    }

    return (
        <form onSubmit={handleSubmit}>
            <input type="text" value={inputState} onChange={handleOnChange}></input>
        </form>
    )
}
AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
export default AddCategory