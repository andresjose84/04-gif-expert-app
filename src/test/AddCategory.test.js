import { shallow } from "enzyme";
import { React } from "react";
import '@testing-library/jest-dom';
import AddCategory from '../components/AddCategory';

describe('Verificar addCategory', () => { 
    const setCategories= jest.fn();
    let wrapper = shallow(<AddCategory setCategories={setCategories}></AddCategory>);

    beforeEach(()=>{
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={setCategories}></AddCategory>);
    });

    test('debe de mostrar correctamente', () => { 
        expect(wrapper).toMatchSnapshot();
    });

    test('debe de cambiar la caja  de texto', () => { 
        
        const input = wrapper.find('input');
        const value = 'Hola mundo';
        input.simulate('change',{ target: {
            value : value
        } });
        expect(wrapper.find('p').text().trim()).toBe(value);

    });

    test('NO debe de postear la informacion on submit', () => { 
    
        wrapper.find('form').simulate('submit',{ preventDefault(){}, target:{value:{}} });
        expect(setCategories).not.toHaveBeenCalled();

    });

    test('debe llamar el setCategories y debe limpiar la caja de texto.', () => { 
        const input = wrapper.find('input');
        const value = 'Batman';
        input.simulate('change',{ target: {
            value : value
        } });
        wrapper.find('form').simulate('submit',{ preventDefault(){}, target:{value:value} });
        expect(setCategories).toHaveBeenCalled();
        expect(setCategories).toHaveBeenCalledTimes(1);
        expect(setCategories).toHaveBeenCalledWith(expect.any(Function));
        expect(wrapper.find('input').prop('value')).toBe('');
    });

});