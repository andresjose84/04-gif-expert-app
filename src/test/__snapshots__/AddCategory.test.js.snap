// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Verificar addCategory debe de mostrar correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
